File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder check = new StringBuilder();
        for(int i =0;i<s.length();i++){
            char ch =s.charAt(i);
            if(ch >=97 && ch<= 122){
                check.append(ch);
            }
            else if(ch>=65 && ch<=90){
                char d=(char)(ch+32);
                check.append(d);
            }
            else if(ch>='0' && ch<='9'){
                check.append(ch);
            }
        }
    int i =0,j=check.length()-1;
    while(i<j){
        if(check.charAt(i)!=check.charAt(j)){return false;}
        i++;
        j--;
    }
    
    return true ;}
}

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int i =0, n= s.length()-1;
        while(i<n){
            if(!isalnum(s[i])){i++;}
            else if(!isalnum(s[n])){n--;}
            else if(tolower(s[i])!=tolower(s[n])){
                return false;
            }
           else{i++,n--;}
           
        }
        return true;
    }
};